### PR TITLE
reworked tests for InteractionTransformer

### PR DESCRIPTION
**Pull request by limlam96**
moved transformer to new testing approach

https://github.com/lvgig/tubular/issues/276

### CODE REVIEW

File: tests/numeric/test_InteractionTransformer.py [hunk 1]
```diff
@@ -1,66 +1,25 @@
...
-                max_degree=2,
-                verbose=True,
-            )
+    @classmethod
+    def setup_class(cls):
+        cls.transformer_name = "InteractionTransformer"
 
     def test_invalid_input_type_errors(self):
         """Test that an exceptions are raised for invalid input types."""
-        with pytest.raises(
```
Comment (Chip2916) [inline tests/numeric/test_InteractionTransformer.py:41]: Q: Do we not still need these? We've kept one of the checks in 'test_invalid_input_type_errors' and can't see that the other two are covered elsewhere?

File: CHANGELOG.rst [hunk 2]
```diff
@@ -16,7 +16,15 @@ Subsections for each version can be one of the following;
...
 Each individual change should have a link to the pull request after the description of the change.
 
-1.3.1 (unreleased)
+1.3.2 (unreleased)
+------------------
+
+Changed
+^^^^^^^
+
+- Refactored tests for InteractionTransformer
```
[the remark this answers is not in the document]
Comment (limlam96) [reply CHANGELOG.rst:25]: the new version here is as result of me forgetting to update the version when I released 1.3.1 haha, so just did here. 

Re the number of changes.. we're kind of just releasing whenever we 'have to' atm, and in this case there was a bug fix in 1.3.1 that we needed for other work, so I made a release quite soon after the previous one

File: tests/numeric/test_InteractionTransformer.py (hunk 1 above)
Comment (limlam96) [reply tests/numeric/test_InteractionTransformer.py:41]: I thinkk these will be covered by GenericInitTests right as that looks at the columns arg?

File: CHANGELOG.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (Chip2916) [reply CHANGELOG.rst:25]: Thanks for explaining :)

**Review by Chip2916 — APPROVED** [review]

**Pull request merged**